packages/material-ui-styles/src/useThemeVariants/useThemeVariants.test.js

```diff
@@ -94,4 +94,48 @@ describe('useThemeVariants', () => {
     const style = window.getComputedStyle(screen.getByTestId('component'));
     expect(style.getPropertyValue('background-color')).to.equal('rgb(255, 255, 0)');
   });
+
+  it('should warn if the used variant is not defined in the theme', function test() {
+    // Warnings are added for whitelisted components, so we need to
+    // test with some name that is in the list, for example MuiButtton
+    const Button = withStyles(
+      {},
+      { name: 'MuiButton' },
+    )((props) => {
+      const { className, ...other } = props;
+      const themeVariantsClasses = useThemeVariants(props, 'MuiButton');
+      return <div className={`${themeVariantsClasses} ${className}`} {...other} />;
+    });
+
+    const theme = createMuiTheme({
+      variants: {
+        MuiButton: [
+          {
+            props: { variant: 'test1' },
+            styles: { backgroundColor: 'rgb(255, 0, 0)' },
+          },
+        ],
+      },
+    });
+
+    expect(() =>
+      render(
+        <ThemeProvider theme={theme}>
+          <Button data-testid="component" variant="test">
+            Test
+          </Button>
+        </ThemeProvider>,
+      ),
+    ).toErrorDev([
+      // strict mode renders twice
+      [
+        `Material-UI: You are using a variant value \`test\` for which you didn't define styles.`,
+        `Please create a new variant matcher in your theme for this variant. To learn more about matchers visit https://material-ui.com/customization/components/#adding-new-component-variants.`,
+      ].join('\n'),
+      [
+        `Material-UI: You are using a variant value \`test\` for which you didn't define styles.`,
+        `Please create a new variant matcher in your theme for this variant. To learn more about matchers visit https://material-ui.com/customization/components/#adding-new-component-variants.`,
```

Comment:
These should be permalinks which is hard to keep track of when working on the docs. We have a special route that contains all the permalinks from error messages:

https://github.com/mui-org/material-ui/blob/4024c2521a9889452d6e8899fd88242eece0c901/docs/public/_redirects#L21-L28

That way older versions of the package still link to valid pages even if we move the page and we can keep the link shorter which is always nice for console output.

Reply:
Nice, I wasn't aware of this. Updated the links. 👍

Reply:
One additional thought. These links will be available on the next.material-ui.com. Should we update the links to be next until that is the prod version?

Reply:
Not the old ones. Otherwise you link to an unstable release channel from a stable version. The new one should link to `next` though (and have a TODO notice to switch to material-ui.com when we release v5).

Reply:
I meant this particular link for this error, as it is for a feature available only in the next version.